### PR TITLE
Update state manager debug view based on design meeting feedback

### PR DESCRIPTION
**Pull request by ajcvickers**
Issue #19109

* Moved to StateManager
* Implemented as ToDebugString on InternalEntityEntry
* Used for ToString on InternalEntityEntry
* Also for ToString on external EntityEntry
* Also added DebugView on InternalEntityEntry to access long form for single entry

### CODE REVIEW

File: src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs [hunk 1]
```diff
@@ -1665,8 +1668,197 @@ public virtual bool IsLoaded([NotNull] INavigation navigation)
...
+        ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+        ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+        ///     any release. You should only use it directly in your code with extreme caution and knowing that
+        ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+        /// </summary>
+        public virtual string ToDebugString(StateManagerDebugStringOptions options)
+        {
+            var builder = new StringBuilder();
+
+            void AppendValue(object value)
```
Comment (AndriySvyryd) [inline src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs]: Local functions go at the end of a method body.

File: src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs [hunk 2]
```diff
@@ -1665,8 +1668,197 @@ public virtual bool IsLoaded([NotNull] INavigation navigation)
...
+
+                    if (currentValue == null)
+                    {
+                        builder.Append("<null>");
+                    }
+                    else if (navigation.IsCollection())
+                    {
+                        builder.Append('[');
+
+                        var relatedEntities = ((IEnumerable)currentValue).Cast<object>().Take(33).ToList();
```
Comment (AndriySvyryd) [inline src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs]: Magic number

File: src/EFCore/ChangeTracking/Internal/StateManagerExtensions.cs [hunk 3]
```diff
@@ -61,193 +57,15 @@ public static class StateManagerExtensions
...
-                        .Append('\'')
-                        .Append(stringValue)
-                        .Append('\'');
-                }
-            }
-
-            void AppendRelatedKey(IEntityType targetType, object value)
+            foreach (var entry in stateManager.Entries
+                .OrderBy(e => e.EntityType.DisplayName())
+                .ThenBy(e => e.BuildCurrentValuesString(e.EntityType.FindPrimaryKey().Properties)))
```
Comment (AndriySvyryd) [inline src/EFCore/ChangeTracking/Internal/StateManagerExtensions.cs]: Even though it's only for debug-time it's still too slow more my eyes. Order by `e.EntityType.Name`, then by `e.EntityType.FindPrimaryKey().Properties` using `PropertyListComparer`

File: src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs (hunk 1 above)
Comment (ajcvickers) [reply src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs]: When did we decide that?

File: src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs (hunk 2 above)
Comment (ajcvickers) [reply src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs]: Will define as a const.

File: src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs (hunk 1 above)
Comment (AndriySvyryd) [reply src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs]: In a design meeting where @smitpatel discussed all the new C# features

Comment (ajcvickers) [reply src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs]: I remember the meeting. I don't remember the decision. @smitpatel @roji What do you remember?

Comment (roji) [reply src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs]: I think I remember we said that, yeah - not sure if it was a hard rule or not. I think it generally helps to make things more readable.

Comment (smitpatel) [reply src/EFCore/ChangeTracking/Internal/InternalEntityEntry.cs]: At the end of the block where it applies.
PascalCase
Static if possible.

**Pull request merged**